lib/enum/basic_enum.rb

```diff
@@ -22,48 +22,78 @@ def number_of_elements_in_collection
   # This one also uses 'count'
   # Try it yourself!
   def number_of_floats_or_fixnums
+    collection.map{|num|
+      if num.is_a? (Integer)
+        num
+      elsif
+        num.is_a? (Float)
+        num
+      else
+        nil
+      end
+    }.compact.count
   end
 
   # Refer to: all?
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-all-3F
+
   def all_words_longer_than_length?(min_length)
+    collection.all?{|word|
+      word.length > min_length}
+
   end
 
 
   # For the next 3 methods refer to: collect or map
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-any-3F
   def contains_a_word_longer_than?(min_length)
+    collection.any?{|word| word.length > min_length}
+
   end
 
   def capitalize_words
+    collection.map{|words|
+      words.capitalize
+    }
   end
 
   def square_numbers
+    collection.map{|num|
+      num ** 2}
   end
 
   # Use symbol to proc notation on this one!
   def reverse_words
+    collection.map(&:reverse)
   end
 
   # Refer to collect_concat or flat_map for these next two
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-flat_map
   # To get the order right for the tests, you'll need to use the absolute value method
   def positive_and_negative_numbers
+    collection.map{|num|
+      num.abs}.flat_map{|num| [num, -num]}
+
   end
 
   # Refer to detect or find
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-detect
   def find_first_awesome_person
+    collection.detect{|awe|
+      awe[:awesome]}
```

Comment:
It would be more accurate to call your local variable `|person|` instead of `|awe|`